front/pages/api/v1/w/[wId]/spaces/[spaceId]/data_sources/[dsId]/folders/index.ts

```diff
@@ -77,5 +77,7 @@ async function handler(
 }
 
 export default withPublicAPIAuthentication(
-  withResourceFetchingFromRoute(handler, "dataSource")
+  withResourceFetchingFromRoute(handler, {
```

Comment:
Can you add a canList in the endpoint as well for the day we remove the auth.isSystemKey :pray: